mani_skill/utils/building/urdf_loader.py

```diff
@@ -1,14 +1,17 @@
 from __future__ import annotations
 
-from typing import TYPE_CHECKING, Any, List, Tuple
+from typing import TYPE_CHECKING, Any, List, Tuple, Optional, Union, Sequence
```

Comment:
remove changes to URDF loader. Instead favor using parse instead so we can use the articulation builders API directly.

also lets modify urdf loader's parse function to return a dictionary for future proofing.

Reply:
Urdf loader changes removed, returning dict from parse(). as a result, needed to

1. change calls to parse to match new api
2. change mjcf loader parse() output for consistency
     - I did not change _mjcf_loader (SapienMJCFLoader), also for consistency since the SapienURDFLoader parse() retvals cannot be altered wtihtout altering sapien itself

Reply:
also, i kept the .load() code the same in the ReplicaCAD scene builder since the load function supports scene_idxs, and the articulation builder will spawn to safe initial poses on its own

I also left the current instances of using .load() untouched